File: Pset 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // get text from user
    string text = get_string("Text: ");

    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    //calculating L and S
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;

// formula to calculate reading index
    int index = round(0.0588 * L - 0.296 * S - 15.8);


// outputting the grade reading level using calculated index
    if (index >= 16) //for indexes greater than zero
    {
        printf("Grade 16+\n");
    }
    else if (index > 0 && index < 16)
    {
        printf("Grade %i\n", (int)index);
    }
    else // for indexes smaller than zero
    {
        printf("Before Grade 1\n");
    }


}

int count_letters(string text)
//function to count number of letters
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i])) //to check if character is alphabetical
        {
            count += 1;
        }
    }
    return count;
}

int count_words(string text)
// function to count number of words
{
    int count = 0;
    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == 32) //to check if character is space
        {
            count += 1;
        }
    }
    return count + 1;
}

int count_sentences(string text)
// function to count number of sentences
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63) //to check if character is either . or ! or ?
        {
            count += 1;
        }
    }
    return count;
}

File: Pset 6/readability.py
# TODO
from cs50 import get_string

# Get string from user
text = get_string("Text: ")
characters = len(text)
letters = sentences = 0
# Set word count to 1 to cater for the last word of the text
words = 1

# Loop through every character of the user's input checking if it is alphabetical or punctuation
for i in text:
    if i.isalpha():
        letters += 1
    # If it is space increment number of words
    elif i == " ":
        words += 1
    # If it is "." or "!" or "?" increment number of sentences
    elif i == "." or i == "!" or i == "?":
        sentences += 1

# Calculate grade and round it up
grade = round(0.0588 * ((letters / words) * 100) - 0.296 * ((sentences / words) * 100) - 15.8)

# Print grade level
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade, {int(grade)}")
